lib/astutils.cpp

```diff
@@ -3142,11 +3142,14 @@ ExprUsage getExprUsage(const Token* tok, int indirect, const Settings* settings)
             return getExprUsage(tok->astParent()->astParent(), indirect, settings);
     }
     if (indirect == 0) {
-        if (Token::Match(tok->astParent(), "%cop%|%assign%|++|--") && !Token::simpleMatch(tok->astParent(), "=") &&
+        if (Token::Match(tok->astParent(), "%cop%|%assign%|++|--") && !Token::Match(tok->astParent(), "=|>>") &&
```

Comment:
This should probably use `isLikelyStreamRead` and should probably check `astIsRHS` as well.

Reply:
See https://github.com/danmar/cppcheck/pull/5315